src/core_plugins/kibana/public/management/sections/settings/advanced_import_controls.js

```diff
@@ -0,0 +1,93 @@
+import { uiModules } from 'ui/modules';
+import { saveAs } from '@spalger/filesaver';
+import { AdvancedImportState } from 'plugins/kibana/management/sections/settings/advanced_import_state.js';
+import advancedImportControlsTemplate from 'plugins/kibana/management/sections/settings/advanced_import_controls.html';
+
+uiModules.get('apps/management')
+  .directive('advancedImportControls', function (config, Notifier, Private) {
+    const notify = new Notifier({ location: 'Advanced Settings' });
+    const advancedImportState = Private(AdvancedImportState);
+
+    return {
+      restrict: 'E',
+      template: advancedImportControlsTemplate,
+      controllerAs: 'advancedImportControls',
+      bindToController: true,
+      controller: class AdvancedImportController {
+        constructor() {
+          this.importState = advancedImportState;
+
+          this.export = () => {
+            const all = config.getAll();
+            // Construct an object with keys that have user-defined values and values
+            // set to those user-defined values
+            const userValues = Object.keys(all).reduce((values, configKey) => {
+              const configObj = all[configKey];
+
+              // This logic is a bit messy. We only want user-defined values that are configurable
+              // so that immediately excludes readonly values
+              if (configObj.readonly) {
+                return values;
+              }
+
+              // Now, we need to make sure they have a user-defined value
+              if (!configObj.hasOwnProperty('userValue')) {
+                return values;
+              }
+
+              // It is also possible that the user-defined value matches
+              // the default value
+              if (configObj.userValue === configObj.value) {
+                return values;
+              }
+
+              values[configKey] = configObj.userValue;
+              return values;
+            }, {});
+            const blob = new Blob([JSON.stringify(userValues)], { type: 'application/json' });
```

Comment:
You could have this output nicely formatted JSON by providing [spacing as a 3rd argument](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify).

`JSON.stringify(userValues, null, 2)`